### PR TITLE
Fixes to GetSummoners and GetSummonersAsync

### PR DESCRIPTION
**Pull request by philippwiddra**
`GetSummoners` and `GetSummonersAsync` accept a maximum of 40 SummonerIds so I added functionality to send multiple requests automatically when there are more than 40 items in the list.
SummonerIds are now implemented as `long` instead of `int`.

### CODE REVIEW

**Comment by BenFradet** [comment]
Very neat idea, I'll have a look later today.

**Comment by BenFradet** [comment]
Would you mind doing the same thing for:
  - [GetLeagues by summoner ids](https://github.com/BenFradet/RiotSharp/blob/develop/RiotSharp/RiotApi.cs#L439,L462) limit of 10 
  - [GetEntireLeagues by summoner ids](https://github.com/BenFradet/RiotSharp/blob/develop/RiotSharp/RiotApi.cs#L470,L494) limit of 10 
  - [GetLeagues by team ids](https://github.com/BenFradet/RiotSharp/blob/develop/RiotSharp/RiotApi.cs#L502,L525) limit of 10 
  - [GetEntireLeagues by team ids](https://github.com/BenFradet/RiotSharp/blob/develop/RiotSharp/RiotApi.cs#L533,L557) limit of 10 
  - [GetSummoners by name](https://github.com/BenFradet/RiotSharp/blob/develop/RiotSharp/RiotApi.cs#L209,L242) limit of 40 
  - [GetMasteryPages](https://github.com/BenFradet/RiotSharp/blob/develop/RiotSharp/RiotApi.cs#L373,L398) limit of 40 
  - [GetRunePages](https://github.com/BenFradet/RiotSharp/blob/develop/RiotSharp/RiotApi.cs#L407,L431) limit of 40 
  - [GetSummonerNames](https://github.com/BenFradet/RiotSharp/blob/develop/RiotSharp/RiotApi.cs#L278,L312) limit of 40 
  - [GetTeams by summoner ids](https://github.com/BenFradet/RiotSharp/blob/develop/RiotSharp/RiotApi.cs#L625,L649) limit of 10 
  - [GetTeams by team ids](https://github.com/BenFradet/RiotSharp/blob/develop/RiotSharp/RiotApi.cs#L657,L678) limit of 10 

Since all those calls have a limit.

**Comment by BenFradet** [comment]
Hey @philippwiddra,
Are you continuing to implement batching for the other methods?
Or, should I just merge this and log another issue for the other endpoints?

**Comment by philippwiddra** [comment]
No problem, I will look into these this weekend.

**Comment by BenFradet** [comment]
Oops, I didn't see you implemented the rest of the limits.
Will review now, sorry.

File: RiotSharpTest/RiotApiTest.cs [hunk 1]
```diff
@@ -48,17 +93,27 @@ public void GetSummonerAsync_ById_Test()
...
         }
 
         [TestMethod]
+        [TestCategory("RiotApi")]
+        public void GetSummoners_ById_Many_Test()
+        {
+            var summoners = api.GetSummoners(batchExampleRegion, summonerIds);
+
+            Assert.IsNotNull(summoners);
+            Assert.IsTrue(summoners.Count > 0);
```
Comment (BenFradet) [inline RiotSharpTest/RiotApiTest.cs]: Shouldn't this be:

```c#
Assert.IsTrue(summoners.Count == summonerIds.Count);
```

Same goes for the other tests.

File: RiotSharp/RiotApi.cs [hunk 2]
```diff
@@ -275,18 +329,28 @@ public SummonerBase GetSummonerName(Region region, int summonerId)
...
-            var children = JsonConvert.DeserializeObject<Dictionary<string, string>>(json);
-            foreach (var child in children)
+
+            foreach (var group in summonerIdGroups)
             {
-                summoners.Add(new SummonerBase(child.Key, child.Value, requester, region));
+                var json = requester.CreateGetRequest(
+                string.Format(SummonerRootUrl, region.ToString()) +
+                    string.Format(NamesUrl, Util.BuildIdsString(group)),
+                region);
```
Comment (BenFradet) [inline RiotSharp/RiotApi.cs]: the indent seems to be off here.

File: RiotSharp/RiotApi.cs [hunk 3]
```diff
@@ -296,18 +360,29 @@ public List<SummonerBase> GetSummonersNames(Region region, List<int> summonerIds
...
-            var children = JsonConvert.DeserializeObject<Dictionary<string, string>>(json);
-            foreach (var sb in children)
+
+            foreach (var group in summonerIdGroups)
             {
-                summoners.Add(new SummonerBase(sb.Key, sb.Value, requester, region));
+                var json = await requester.CreateGetRequestAsync(
+                string.Format(SummonerRootUrl, region.ToString()) +
+                    string.Format(NamesUrl, Util.BuildIdsString(group)),
+                region);
```
Comment (BenFradet) [inline RiotSharp/RiotApi.cs]: same here

**Comment by BenFradet** [comment]
LGTM! 

Except for a few things:

- AFAICT, you haven't done it for the [leagues by team ids endpoints](https://github.com/philippwiddra/RiotSharp/blob/develop/RiotSharp/RiotApi.cs#L713,L768), is there a reason why not?
- the unit tests could be bit better by testing that we get more results than the limit
- there are a few formatting quirks

**Comment by philippwiddra** [comment]
The problem with `Assert.IsTrue(summoners.Count == summonerIds.Count);` is that some SummonerIds may no longer exist and therefor will not be found. So there will most probably be a discrepency between the numbers.
Maybe we could load a list of summonerIds from somewhere else (like the challenger-league) and use that result as guaranteed existing summonerIds.
This would apply to all limit-tests.

I couldn't really test the teamIds thing, because somehow the Riot-API does not work properly with teamIds. I guessed your tests where set to `[Ignore]` because of that.
Should I implement them anyway and just assume the team-endpoints won't change?

I'll fix the formatting to something more readable soon.

**Comment by BenFradet** [comment]
Couldn't we do:

```c#
Assert.IsTrue(summoners.Count > limit);
```

then?

Regarding, the leagues by team ids endpoint, yes, please, go ahead and add `[Ignore]` to your tests. Progress regarding this issue is tracked in #205 and will be fixed in time.

**Comment by BenFradet** [comment]
@philippwiddra any updates on this?

**Pull request closed without merging**